src/autoscaler/api/policyvalidator/policy_validator.go

```diff
@@ -0,0 +1,369 @@
+
+	rootContext := gojsonschema.NewJsonContext("(root)", nil)
+
+	//check InstanceMinCount and InstanceMaxCount
+	if policy.InstanceMin >= policy.InstanceMax {
+		instanceMinContext := gojsonschema.NewJsonContext("instance_min_count", rootContext)
+		errDetails := gojsonschema.ErrorDetails{
+			"instance_min_count": policy.InstanceMin,
+			"instance_max_count": policy.InstanceMax,
+		}
+		formatString := "instance_min_count {{.instance_min_count}} is higher or equal to instance_max_count {{.instance_max_count}}"
+		err := newPolicyValidationError(instanceMinContext, formatString, errDetails)
+		result.AddError(err, errDetails)
+	}
+
+	// check InstanceMinCount, IntanceMaxCount and InitialInstanceMinCount for schedules
+	if policy.Schedules == nil {
+		return
+	}
+	schedulesContext := gojsonschema.NewJsonContext("schedules", rootContext)
+
+	pv.validateRecurringSchedules(policy, schedulesContext, result)
+	pv.validateSpecificDateSchedules(policy, schedulesContext, result)
+}
+
+func (pv *PolicyValidator) validateRecurringSchedules(policy *models.ScalingPolicy, schedulesContext *gojsonschema.JsonContext, result *gojsonschema.Result) {
+	recurringScheduleContext := gojsonschema.NewJsonContext("recurring_schedule", schedulesContext)
+	for i, recSched := range policy.Schedules.RecurringSchedules {
+		if recSched.ScheduledInstanceMin >= recSched.ScheduledInstanceMax {
+			instanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.instance_min_count", i), recurringScheduleContext)
+			errDetails := gojsonschema.ErrorDetails{
+				"instance_min_count": recSched.ScheduledInstanceMin,
+				"instance_max_count": recSched.ScheduledInstanceMin,
+			}
+			formatString := "instance_min_count {{.instance_min_count}} is higher or equal to instance_max_count {{.instance_max_count}}"
+			err := newPolicyValidationError(instanceMinContext, formatString, errDetails)
+			result.AddError(err, errDetails)
+		}
+
+		if recSched.ScheduledInstanceInit > recSched.ScheduledInstanceMax {
+			initialInstanceMinContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d.initial_min_instance_count", i), recurringScheduleContext)
+			errDetails := gojsonschema.ErrorDetails{
+				"initial_min_instance_count": recSched.ScheduledInstanceInit,
+				"instance_max_count":         recSched.ScheduledInstanceMin,
+			}
+			formatString := "initial_min_instance_count {{.initial_min_instance_count}} is greater than instance_max_count {{.instance_max_count}}"
+			err := newPolicyValidationError(initialInstanceMinContext, formatString, errDetails)
+			result.AddError(err, errDetails)
+		}
+
+		//start_time should be before end_time
+		if compareTimesGTEQ(recSched.StartTime, recSched.EndTime) {
+			currentRecSchedContext := gojsonschema.NewJsonContext(fmt.Sprintf("%d", i), recurringScheduleContext)
+			errDetails := gojsonschema.ErrorDetails{
+				"i": i,
+			}
+			formatString := "start_time is after end_time"
```

Comment:
considering the function compareTimes**GTEQ**,  the msg here should be "same or after"